### PR TITLE
[CALCITE-5593] Support aliases query in Elasticsearch adapters 

### PR DESCRIPTION
**Pull request by coolbeevip**
Jira issue https://issues.apache.org/jira/browse/CALCITE-5593

### CODE REVIEW

File: elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchema.java [hunk 1]
```diff
@@ -120,7 +120,9 @@ private Set<String> indicesFromElastic() throws IOException {
         throw new IllegalStateException(message);
       }
 
+      Set<String> aliases = Sets.newHashSet(root.findPath("aliases").fieldNames());
       Set<String> indices = Sets.newHashSet(root.fieldNames());
+      indices.addAll(aliases);
```
Comment (libenchao) [inline elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchema.java:125]: I'm not familiar with es adapter, but is it possible to add a unit test to show that it only works after this change?

Comment (coolbeevip) [reply elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchema.java:125]: Sure, I need to understand the current ES integration testing method first, I am only familiar with ES testing based on testcontainers

Comment (coolbeevip) [reply elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchema.java:125]: Hi,   I added a method to create an alias and added unit tests

zips_alias is an alias of zips

```
@Test void testAlias() {
    calciteAssert()
        .query("select state, city from elastic.zips_alias");
  }
```

**Comment by julianhyde** [comment]
The commit message will become the release notes. It needs to describe the feature. Currently it doesn't mention that it concerns the Elasticsearch adatper.

**Comment by coolbeevip** [comment]
> The commit message will become the release notes. It needs to describe the feature. Currently, it doesn't mention that it concerns the Elasticsearch adapter.

thanks for the reminder, I have fixed the commit message

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java [hunk 2]
```diff
@@ -189,6 +191,11 @@ private CalciteAssert.AssertThat calciteAssert() {
         .returnsCount(0);
   }
 
+  @Test void testAlias() {
+    calciteAssert()
+        .query("select state, city from elastic.zips_alias");
```
Comment (libenchao) [inline elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java:196]: This test seems asserting nothing, hence it won't be run actually.

**Comment by libenchao** [comment]
@coolbeevip The CI failed now, it seems that you are using an guava feature which is not available in our oldest supported guava version: 16.0.1

**Comment by coolbeevip** [comment]
> @coolbeevip The CI failed now, it seems that you are using an guava feature which is not available in our oldest supported guava version: 16.0.1
Thanks for the tip, I will use Java StreamSupport.stream instead Guava Streams.stream

**Review by libenchao — APPROVED** [review]
+1, merging. @coolbeevip Thanks for your contribution.

**Pull request closed without merging**